simde/x86/avx512/roundscale.h

```diff
@@ -0,0 +1,41 @@
+#if !defined(SIMDE_X86_AVX512_ROUNDSCALE_H)
+#define SIMDE_X86_AVX512_ROUNDSCALE_H
+
+#include "types.h"
+#include "../sse.h"
+#include "cmp.h"
+
+HEDLEY_DIAGNOSTIC_PUSH
+SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
+SIMDE_BEGIN_DECLS_
+
+#if defined(SIMDE_X86_AVX512VL_NATIVE) && defined(SIMDE_X86_AVX512F_NATIVE)
+  #define simde_mm_roundscale_ps(a, imm8) _mm_roundscale_ps((a), (imm8))
+#else
+  SIMDE_FUNCTION_ATTRIBUTES
+  simde__m128
+  simde_mm_roundscale_ps (simde__m128 a, int imm8)
+      SIMDE_REQUIRE_CONSTANT_RANGE(imm8, 0, 255) {
+    simde__m128 temp, r, clear_sign;
+    int32_t
+      exponent1 = (127 + ((imm8 >> 4) & 15)) << 23,
+      exponent2 = (127 - ((imm8 >> 4) & 15)) << 23;
+
+    SIMDE_CONSTIFY_16_(simde_mm_round_ps, temp, simde_mm_setzero_ps(), imm8 & 15, simde_mm_mul_ps(a, simde_mm_set1_ps(*HEDLEY_REINTERPRET_CAST(SIMDE_FLOAT32_TYPE*, &exponent1))));
+    r = simde_mm_mul_ps(simde_mm_set1_ps(*HEDLEY_REINTERPRET_CAST(SIMDE_FLOAT32_TYPE*, &exponent2)), temp);
+
+    clear_sign = simde_mm_andnot_ps(simde_mm_set1_ps(SIMDE_FLOAT32_C(-0.0)), r);
+    r = simde_mm_mask_mov_ps(r, simde_mm_cmp_ps_mask(clear_sign, simde_mm_set1_ps(SIMDE_MATH_INFINITY), SIMDE_CMP_EQ_OQ), a);
```

Comment:
`simde_mm_cmp_ps_mask` and `simde_mm_mask_mov_ps` are only available in AVX-512, and emulating them is pretty slow.  Instead, how about using `simde_mm_cmpeq_ps` and `simde_x_mm_select_ps`?  They'll perform a lot better on machines without AVX-512, *especially* on other architectures.

Other than that, this looks really good.

Reply:
Will update using cmpeq and select functions